### PR TITLE
Bug fixes around var watching

### PR DESCRIPTION
**Pull request by ericdevg**
#208 #178

@AndreSayreMSFT

### CODE REVIEW

File: PowershellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingService.cs [hunk 1]
```diff
@@ -269,6 +269,15 @@ public Collection<Variable> GetScopedVariable()
 
                     if (psVar != null)
                     {
+                        if (psVar.Value is PSObject &&
+                            !(((PSObject)psVar.Value).ImmediateBaseObject is PSCustomObject))
+                        {
```
Comment (ASayre) [inline PowershellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingService.cs:274]: Do we ever have to worry about further nesting?  eg, can the immediateBaseObject be another PSObject?  What if things are piped around and minuplated?  Or is it only ever one layer deep?

File: PowershellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingService.cs [hunk 2]
```diff
@@ -387,13 +396,20 @@ public Collection<Variable> GetObjectVariable(string varName)
...
+                            string key = string.Format("{0}\\{1}", varName, propertyInfo.Name);
+                            _propVariables[key] = val;
+                        }
+                    }
+                    catch (Exception ex)
                     {
-                        string key = string.Format("{0}\\{1}", varName, propertyInfo.Name);
-                        _propVariables[key] = val;
+                        ServiceCommon.Log("Property infomation is not able to be retrieved through refletion due to exception: {0}", ex.Message);
```
Comment (ASayre) [inline PowershellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingService.cs]: nit: "refletion"

File: PowershellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingService.cs (hunk 1 above)
Comment (ASayre) [inline PowershellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingService.cs:274]: eg, why ImmediateBaseObject instead of BaseObject?

Comment (ericdevg) [inline PowershellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingService.cs:274]: Good question, actually I guess the ImmediateBaseObject's definition should explain it: "Gets the encapsulated base object. If the base object is another PSObject object, then its base object is returned."

File: PowershellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingService.cs (hunk 2 above)
Comment (ericdevg) [inline PowershellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingService.cs]: Fixed.

**Pull request merged**